clang/lib/Driver/Compilation.cpp

```diff
@@ -99,7 +99,7 @@ static void HandleXarchArgs(DerivedArgList *OffloadArgList, const Driver &D,
               OffloadArgList->MakeArgStringRef(XarchVecs[Index]));
       }
     } else
-      XarchValueRefs.push_back(StringRef(XarchV));
+      XarchValueRefs.push_back(OffloadArgList->MakeArgString(XarchV));
```

Comment:
Any reason why `OffloadArgList->MakeArgStringRef` is not used instead?

Reply:
Hi, @srividya-sundaram 
Both MakeArgStringRef and MakeArgString returns a "const char *" value and MakeArgString just calls "MakeArgStringRef" internally, so I just picked one of them without any special reason.
Thanks very much.